addon/components/discover-page/component.js

```diff
@@ -0,0 +1,808 @@
+                    this.set('type', this.get('activeFilters.types').join('OR'));
+                } else {
+                    this.set('type', '');
+                }
+            }
+        },
+        removeFilter(type, filterValue) {
+            // Ember-SHARE action.  Could be used to remove filters in Active Filters box (when Ember-SHARE and PREPRINTS/REGISTRIES code here is integrated)
+            let currentValue = getSplitParams(this.get(type)) || [];
+            let index = currentValue.indexOf(filterValue);
+            if (index > -1) {
+                currentValue.splice(index, 1);
+            }
+            currentValue = currentValue.length ? encodeParams(currentValue) : '';
+            this.set(type, currentValue);
+            this.get('facetFilters');
+        },
+        search() {
+            // Only want to track search here when button clicked. Keypress search tracking is debounced in trackSearch
+            Ember.get(this, 'metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: 'Discover - Search',
+                    extra: this.get('q')
+                });
+
+            this.search();
+        },
+        selectSortOption(option) {
+            // Runs when sort option changed in dropdown
+            this.set('sort', option);
+            Ember.get(this, 'metrics')
+                .trackEvent({
+                    category: 'dropdown',
+                    action: 'select',
+                    label: `Sort by: ${option || 'relevance'}`
+                });
+            this.search();
+        },
+        setLoadPage(pageNumber) {
+            // Adapted from PREPRINTS for pagination. When paginating, sets page and scrolls to top of results.
+            this.set('page', pageNumber);
+            if (scroll) {
+                this.scrollToResults();
+            }
+            this.loadPage();
+        },
+        toggleShowLuceneHelp() {
+            // Toggles display of Lucene Search help modal
+            this.toggleProperty('showLuceneHelp');
+        },
+        typing(val, event) {
+            /**
+             * Fires on keyup in search bar.
+             *
+             * Ignores all keycodes that don't result in the value changing
+             * 8 == Backspace, 32 == Space
+             */
+            if (event.keyCode < 49 && !(event.keyCode === 8 || event.keyCode === 32)) {
+                return;
+            }
+            // Tracks search on keypress, debounced
+            Ember.run.debounce(this, this.trackDebouncedSearch, 3000);
+            this.search();
+        },
+        updateFilters(filterType, item) {
+            // For PREPRINTS and REGISTRIES.  Modifies activeFilters.
+            item = typeof item === 'object' ? item.text : item;
+            const filters = Ember.$.extend(true, Ember.A(), this.get(`activeFilters.${filterType}`));
```

Comment:
Even though this deep copies, I also need to put it into an Ember Array here so I can do push/removeObject shortly after.